python/pyspark/cloudpickle.py

```diff
@@ -40,164 +40,126 @@
 NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 """
-
+from __future__ import print_function
 
 import operator
 import os
+import io
 import pickle
 import struct
 import sys
 import types
 from functools import partial
 import itertools
-from copy_reg import _extension_registry, _inverted_registry, _extension_cache
-import new
 import dis
 import traceback
-import platform
-
-PyImp = platform.python_implementation()
-
 
-import logging
-cloudLog = logging.getLogger("Cloud.Transport")
```

Comment:
Would you be amenable to moving cloudpickle to a separate repository? We'd like to be able to rely on it in IPython parallel as well as other projects. In the past couple days, folks at the PyCon sprints have been adding tests for the [current codebase](https://github.com/cloudpipe/cloudpickle).